### PR TITLE
feat(auth): Adds confirmSignIn api and operation

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:*

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/RespondToAuthChallenge.swift [hunk 1]
```diff
@@ -36,4 +36,9 @@ extension RespondToAuthChallenge {
                                        attributeKey: nil)
     }
 
+    var debugDictionary: [String: Any] {
+        return ["challenge": challenge,
+                "username": username]
```
Comment (harsh62) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/RespondToAuthChallenge.swift:41]: should some of the sensitive information be masked?

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    override public func main() {
+        if isCancelled {
+            finish()
+            return
+        }
+        authStateMachine.getCurrentState { [weak self] in
+            guard case .configured(let authenticationState, _) = $0,
+                  case .signingIn(let signInState) = authenticationState else {
+                // TODO: Return proper error
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift:46]: yes, there will be a follow up

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+            switch signInState {
+            case .resolvingSMSChallenge(let challengeState):
+                guard case .waitingForAnswer = challengeState else {
+                    // TODO: Return proper error
+                    return
+                }
+                self?.sendConfirmSignInEvent()
+            default:
+                print("")
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift:59]: Added TODO

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            }
+            guard case .configured(let authNState, _) = $0 else {
+                return
+            }
+            switch authNState {
+
+            case .signedIn(let signedInData):
+                let cognitoTokens = signedInData.cognitoUserPoolTokens
+                self.storeUserPoolTokens(cognitoTokens)
+
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift]: Added

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        }
+    }
+
+    private func sendStoreCredentialsEvent(with userPoolTokens: AWSCognitoUserPoolTokens) {
+        let credentials = AmplifyCredentials(userPoolTokens: userPoolTokens,
+                                             identityId: nil,
+                                             awsCredential: nil)
+        let event = CredentialStoreEvent.init(eventType: .storeCredentials(credentials))
+        credentialStoreStateMachine.send(event)
+    }
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AmplifyConfirmSignInOperation.swift:146]: Yes, next followup PR is for refactor

**Review by harsh62 — APPROVED** [review]

**Pull request merged**